app/models/course/group.rb

```diff
@@ -15,6 +15,57 @@ class Course::Group < ActiveRecord::Base
   validate :validate_new_users_are_unique
   validate :validate_presence_of_group_manager, on: :update
 
+  # @!attribute [r] average_experience_points
+  #   Returns the average experience points of group users in this group who are students.
+  calculated :average_experience_points, (lambda do
+    Course::GroupUser.where { group_id == course_groups.id }.
+      joins { course_user.experience_points_records.outer }.
+      where { course_user.role == CourseUser.roles[:student] }.
+      # CAST is used to force a float division (integer division by default).
+      # greatest(#, 1) is used to avoid division by 0.
+      select(<<-SQL)
```

Comment:
Tried it, doesn't work though. The issue is the syntax of `cast` - only methods I saw dealing with casting in rails >_<